### PR TITLE
Infra feat client foraging

### PR DESCRIPTION
**Pull request by Eirikalb**
# Summary

Adds client implementation of basic foraging agent, that simply contributes resources randomly between 0 and 100.

Also adds some shared types to help organize foraging code and adds a server `runForage()` method as well
## Additional Information

**I am having white space issues plz help!**

## Test Plan

- Add information on how you tested your changes.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -35,4 +35,4 @@ jobs:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       with:
         build_dir: website/build
-        branch: gh-pages
\ No newline at end of file
+        branch: gh-pages
```
Comment (lhl2617) [inline .github/workflows/deploy.yml]: nani

**Review by lhl2617 — APPROVED** [review]
in eirik i trust -- fix those newline stuff and merge

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (lhl2617) [reply .github/workflows/deploy.yml]: try running
```
git add --renormalize .
```

Comment (Eirikalb) [reply .github/workflows/deploy.yml]: I just removed the newlines in those files manually

**Pull request merged**